auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -676,6 +758,19 @@ private Path tempPath() {
         public void create() throws RealmUnavailableException {
             if (System.getSecurityManager() == null) {
                 createPrivileged();
+                if(privateKey != null) {
+                    try {
+                        createDigitalSignature(path.toString(), this.name);
+                        writeToMainIndex(this.mainIndex, this.path, this.secretKey);
+                    } catch (IllegalStateException e) {
+                        throw ElytronMessages.log.unableToGenerateSignature(path.toString());
+                    } catch (IOException | GeneralSecurityException e) {
+                        throw ElytronMessages.log.unableToAccessMainIndex(this.mainIndex.toString());
+                    }
+                    if (!verifyIntegrity()) {
```

Comment:
It looks like this is verifying the integrity of the identity immediately after writing it. Is that intended?